_posts/2020-06-23-Durr_Hoyer_Post_1.md

```diff
@@ -0,0 +1,328 @@
+I am programming the algorithm in Q# with a python host script.
+The circuit's gates will be analyzed and some code is given.  
+https://github.com/mridulsar/DurrHoyerLibrary   
+
+### My Background
+I began my journey from math to computers through my passion for ML algorithms, where I stumbled upon quantum computing. 
+I was instantly pulled in by the elegant algorithms and proposed efficiency.
+Recently, I started looking into quantum machine learning where you can model nodes of neural networks as qubits. 
+I took a deep dive into the research papers and found myself amazed by what was out there.
+I began with a Coursera course from St. Petersburg University on quantum computing with detail on quantum algorithm design and quantum computer architecture. 
+From here I tinkered with Q# and developed basic algorithms that I previously learned.
+I found Dürr and Høyer's paper and saw it as an fusion of Duestch's, Shor's, and Grover's Algorithm; a perfect next step for me.
+
+### Conclusion
+
+So far I have been greatly enjoying throwing myself into this new and exciting world of quantum computing. 
+I have seen some awesome principles translate across quantum algorithms that are keeping me engaged.
+In specific, I mentioned I felt Dürr and Høyer used principles from Duestch's, Grover's, and Shor's algorithm.
+Duestsch's Algorithm famously simplifies a classical problem into an non intuitive oracle function as done in Dürr and Høyer's algorithm to mark all states that satisfy T[j]<T[y]. 
+Grover's Algorithm is quite literally applied in this algorithm, though a generalized version is used.
+Lastly, Shor's Algorithm shows off the power of combining classical and quantum systems to achieve outstanding results, which is shown in Dürr and Høyer's algorithm by bounding our time, a classical step in the algorithm.
+I love seeing such fundamental concepts continue to push boundaries.  
+
+This algorithm should be available to use within Q# within the next few weeks once testing is done to ensure usability. 
+QESA is really the algorithm of interest, to get a running version of QMSA is just to show that this algorithm does indeed work. 
+What is more interesting is creating variations of QMSA to fit your needs while utilizing QESA. 
+It is important to note that currently QESA can only solve for one unique solution. 
+In the future, QESA will be more robust and multiple solutions can be found.
+In the mean time you can implement this code for yourself, I would love to see what this community can do with an algorithm like this.
+If you had any suggestions or questions feel free to send me an email.
+
+https://github.com/mridulsar/DurrHoyerLibrary   
+
+------------------
+
+## My current challenges in this project:   
+
+1. Implementing search for multiple solutions.    
+
+2. Developing alleviation for Big-O in a fluid manner.   

```

Comment:
Not quite sure what this means, could you clarify?

Reply:
for 1. I am thinking of extending the python host script functionality to allow for finding multiple solutions using grovers for multiple solutions
for 2. I am thinking the user can manually adjust the time limit of the python host script as mentioned in my response to the conversation above this for slower computers.